packages/jest-runtime/src/index.ts

```diff
@@ -776,7 +778,46 @@ class Runtime {
       return this._environment.global.process;
     }
 
-    return require(moduleName);
+    const module = require(moduleName);
```

Comment:
Might want to check `_coreModulesProxyCache` first (see below) before possibly needlessly calling `require`.